dev/src/watch.ts

```diff
@@ -27,6 +27,11 @@ import {ResourcePath} from './path';
 import {requestTag} from './util';
 import {DocumentSnapshot} from './document';
 import {DocumentChange} from './document-change';
+import {PassThrough} from 'stream';
```

Comment:
Thanks for tackling this file. I would like to also be much more pedantic here about the types, but I think that's beyond the scope of a simple TS conversion. If you'd like, I can enforce stricter types once this PR is merged.